### PR TITLE
Admin: Enable order cancelation in Order edit view (SHOOP-2401)

### PR DESCRIPTION
**Pull request by shawnadelic**
Enable cancelation in admin Order edit view for unpaid and unshipped
orders.

Refs SHOOP-2401

### CODE REVIEW

File: shoop/admin/modules/orders/views/detail.py [hunk 1]
```diff
@@ -17,17 +17,28 @@
...
 from shoop.admin.utils.urls import get_model_url
 from shoop.apps.provides import get_provide_objects
-from shoop.core.models import Order, OrderStatus, OrderStatusRole
+from shoop.core.models import (
+    Order, OrderStatus, OrderStatusRole, ShippingStatus
+)
 from shoop.utils.excs import Problem
 
 
+def _can_set_canceled(order):
```
Comment (tulimaki) [inline shoop/admin/modules/orders/views/detail.py]: This logic should be in ``Order`` model

**Comment by tulimaki** [comment]
Can you add changelog also for core changes. Consider separating core and admin changes in to different commits.

**Comment by shawnadelic** [comment]
test this please

File: shoop/core/models/_orders.py [hunk 2]
```diff
@@ -554,6 +554,15 @@ def can_set_complete(self):
...
         return (not self.is_complete()) and fully_shipped and (not canceled)
 
+    def is_canceled(self):
+        return (self.status.role == OrderStatusRole.CANCELED)
+
+    def can_set_canceled(self):
+        canceled = (self.status.role == OrderStatusRole.CANCELED)
+        paid = self.is_paid()
+        shipped = (self.shipping_status != ShippingStatus.NOT_SHIPPED)
+        return (not canceled) and (not paid) and (not shipped)
```
Comment (tulimaki) [inline shoop/core/models/_orders.py]: I think this would be more readable with ``not (canceled or paid or shipped)``

**Pull request merged**